include/RevProc.h

```diff
@@ -120,41 +120,50 @@ class RevProc{
     uint64_t cyclesIdle_MemoryFetch;
   };
 
+  ///< RevProc: Returns this proc's stats
   RevProcStats GetStats() { Stats.memStats = mem->memStats; return Stats; }
 
+  ///< RevProc: Returns a pointer to the RevMem object
   RevMem& GetMem() const { return *mem; }
 
-  // Get the bitset of state changes
-  const std::bitset<_REV_HART_COUNT_>& GetThreadStateChanges() { return ThreadStateChanges; }
-
-  // Zeros the bitset of state changes
-  void ClearThreadStateChanges(){ ThreadStateChanges.reset(); }
+  ///< RevProc: Get this Proc's feature
+  RevFeature* GetRevFeature() const { return feature; }
 
-  /// RevProc: SpawnThread creates a new thread and returns its ThreadID
-  void CreateThread(uint32_t NewTid, uint64_t fn, void* arg);
+  ///< RevProc: Mark a current request as complete
+  void MarkLoadComplete(const MemReq& req);
 
-  /// RevProc: Returns the current HartToExec active pid
-  uint32_t GetActiveThreadID();
+  ///< RevProc: Get pointer to Load / Store queue used to track memory operations
+  std::shared_ptr<std::unordered_map<uint64_t, MemReq>> GetLSQueue(){ return LSQueue; }
 
-  /// RevProc: Queue of threads to be spawned (RevProc creates the thread, RevCPU readies it for execution)
-  const std::queue<std::shared_ptr<RevThread>>& GetNewThreadInfo() { return NewThreadInfo; }
+  ///< RevProc: Finds a Hart that is available to assign the Thread (already inside this RevProc's AssignedThreads) 
+  void AssignThread(const uint32_t& ThreadID);
 
-  ///< RevProc: Holds the info for all new threads to be spawned
-  ///           Right now this is through the following functions:
-  ///           - rev_pthread_create
-  std::queue<std::shared_ptr<RevThread>> NewThreadInfo;
+  /// RevProc: Returns the RevThread that is on the given Hart
+  /// TODO:(BEFORE MERGE): I'm not in love with the fact this isn't safe (you first must check if the Hart has a thread via HartHasThread)
+  const std::shared_ptr<RevThread>& ThreadOnHart(uint16_t HartID){ 
+    return AssignedThreads.at(Harts.at(HartID)->GetAssignedThreadID()); 
+  }
 
   ///< RevProc: Used for scheduling in RevCPU (if Utilization < 1, there is at least 1 unoccupied HART )
-  double GetHartUtilization() const { return (AssignedThreads.size() * 100.0) / _REV_HART_COUNT_; }
+  double GetHartUtilization() const { return (AssignedThreads.size() * 100.0) / Harts.size(); }
 
-  ///< RevProc: Get this Proc's feature
-  RevFeature* GetRevFeature() const { return feature; }
+  // Get the hart ids of the threads that changed state
+  std::queue<std::shared_ptr<RevThread>>& GetThreadsThatChangedState() { return ThreadsThatChangedState; }
```

Comment:
Probably should be `const std::queue<std::shared_ptr<RevThread>>& GetThreadsThatChangedState()` unless the returned reference is intended to be mutable. Can be abbreviated `const auto& GetThreadsThatChangedState()` for brevity.

Reply:
It needs to be mutable because as the RevCPU handles each thread in there based on their change of state it pops the thread from this queue.